logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
what is going to be used? this line explicitly or you are generally speaking for the whole class?  ... it is currently in use in my other branch. I am doing this to simplify PRs review and try to not mix concerns. this is a PR about adding support the the lock release.